app/src/main/java/com/kickstarter/viewmodels/PrivacyViewModel.kt

```diff
@@ -50,91 +51,97 @@ interface PrivacyViewModel {
         private val userOutput = BehaviorSubject.create<User>()
-        private val updateSuccess = PublishSubject.create<Void>()
+        private val updateSuccess = PublishSubject.create<Unit>()
 
         private val unableToSavePreferenceError = PublishSubject.create<Throwable>()
 
-        val inputs: PrivacyViewModel.Inputs = this
-        val outputs: PrivacyViewModel.Outputs = this
-        val errors: PrivacyViewModel.Errors = this
+        val inputs: Inputs = this
+        val outputs: Outputs = this
+        val errors: Errors = this
 
-        private val client = requireNotNull(environment.apiClient())
-        private val currentUser = requireNotNull(environment.currentUser())
+        private val client = requireNotNull(environment.apiClientV2())
+        private val currentUser = requireNotNull(environment.currentUserV2())
+        private val disposables = CompositeDisposable()
 
         init {
             this.client.fetchCurrentUser()
                 .retry(2)
-                .compose(Transformers.neverError())
-                .compose(bindToLifecycle())
+                .compose(Transformers.neverErrorV2())
                 .subscribe { this.currentUser.refresh(it) }
+                .addToDisposable(disposables)
 
             val currentUser = this.currentUser.observable()
 
             currentUser
                 .take(1)
-                .filter { it.isNotNull() }
-                .compose(bindToLifecycle())
-                .subscribe { this.userOutput.onNext(it) }
+                .filter { it.isNotNull() && it.getValue().isNotNull() }
+                .map { it.getValue() }
+                .subscribe { this.userOutput.onNext(it!!) }
+                .addToDisposable(disposables)
 
             currentUser
-                .compose(bindToLifecycle())
-                .filter { it.isNotNull() }
-                .map { user -> user.createdProjectsCount().isNonZero() }
-                .subscribe(this.hidePrivateProfileRow)
+                .filter { it.isNotNull() && it.getValue().isNotNull() }
+                .map { user -> user.getValue()?.createdProjectsCount().isNonZero() }
+                .subscribe { this.hidePrivateProfileRow.onNext(it) }
+                .addToDisposable(disposables)
 
             val updateSettingsNotification = this.userInput
                 .concatMap { this.updateSettings(it) }
 
             updateSettingsNotification
-                .compose(values())
-                .compose(bindToLifecycle())
+                .compose(valuesV2())
                 .subscribe { this.success(it) }
+                .addToDisposable(disposables)
 
             updateSettingsNotification
-                .compose(errors())
-                .compose(bindToLifecycle())
-                .subscribe(this.unableToSavePreferenceError)
+                .compose(errorsV2())
+                .filter { it.isNotNull() }
+                .subscribe { this.unableToSavePreferenceError.onNext(it!!) }
```

Comment:
we try to avoid the not-null forced assertion operator `!!`. we could use `.map { requireNotNull(it) }` after the `filter` instead